packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -684,8 +686,9 @@ export function fid(url: string | undefined | null): string | null | undefined {
     if (isNull(url)) {
         return null;
     }
+    const { renderMode } = vmBeingRendered;
     // Apply transformation only for fragment-only-urls, and only in shadow DOM
-    if (/^#/.test(url) && isLightRenderModeVM(vmBeingRendered!)) {
+    if (/^#/.test(url) && renderMode === RenderMode.Shadow) {
```

Comment:
Shouldn't this run on for synthetic shadow DOM?

Reply:
That's the next PR :)